libs/langchain/langchain/schema/runnable/base.py

```diff
@@ -1437,6 +1444,41 @@ def invoke(
         return self._call_with_config(self.func, input, config)
 
 
+class RunnableEach(Serializable, Runnable[List[Input], List[Output]]):
+    """
+    A runnable that delegates calls to another runnable with each element of the input sequence.
+    """
+
+    bound: Runnable[Input, Output]
+
+    class Config:
+        arbitrary_types_allowed = True
+
+    @property
+    def lc_serializable(self) -> bool:
+        return True
+
+    @property
+    def lc_namespace(self) -> List[str]:
+        return self.__class__.__module__.split(".")[:-1]
+
+    def map(self) -> RunnableEach[Input, Output]:  # type: ignore[override]
+        return self
```

Comment:
I left a comment on the earlier thread about this, but since the code has moved that comment thread disappeared. So I'm leaving this as a reminder to myself and others :)

Reply:
I don't want more layers of abstraction here, we have a history of overdoing that in this library. If you guys really want to break actual functionality for the sake of a linter I'll just make RunnableEach.map raise NotImplemented

But IMO that kind of decision where we're actively making the library worse to satisfy a linter (no matter how academically pleasing this feature of the linter may be) is incorrect. In any case I'll let you guys decide

Reply:
Sure